library/src/main/java/com/novoda/downloadmanager/lib/DownloadThread.java

```diff
@@ -845,14 +863,40 @@ private void addRequestHeaders(State state, HttpURLConnection conn) {
     private void notifyDownloadCompleted(State state, int finalStatus, String errorMsg, int numFailed) {
         notifyThroughDatabase(state, finalStatus, errorMsg, numFailed);
         if (DownloadStatus.isCompleted(finalStatus)) {
-            originalDownloadInfo.broadcastIntentDownloadComplete(finalStatus);
+            broadcastIntentDownloadComplete(finalStatus);
         } else if (DownloadStatus.isInsufficientSpace(finalStatus)) {
-            originalDownloadInfo.broadcastIntentDownloadFailedInsufficientSpace();
+            broadcastIntentDownloadFailedInsufficientSpace();
         }
     }
 
+    public void broadcastIntentDownloadComplete(int finalStatus) {
+        Intent intent = new Intent(DownloadManager.ACTION_DOWNLOAD_COMPLETE);
+        intent.setPackage(getPackageName());
+        intent.putExtra(DownloadManager.EXTRA_DOWNLOAD_ID, originalDownloadInfo.getId());
+        intent.putExtra(DownloadManager.EXTRA_DOWNLOAD_STATUS, finalStatus);
+        intent.setData(originalDownloadInfo.getMyDownloadsUri());
+        if (originalDownloadInfo.getExtras() != null) {
+            intent.putExtra(EXTRA_EXTRA, originalDownloadInfo.getExtras());
+        }
+        context.sendBroadcast(intent);
+    }
+
+    public void broadcastIntentDownloadFailedInsufficientSpace() {
+        Intent intent = new Intent(DownloadManager.ACTION_DOWNLOAD_INSUFFICIENT_SPACE);
+        intent.setPackage(getPackageName());
+        intent.putExtra(DownloadManager.EXTRA_DOWNLOAD_ID, originalDownloadInfo.getId());
+        intent.setData(originalDownloadInfo.getMyDownloadsUri());
+        if (originalDownloadInfo.getExtras() != null) {
+            intent.putExtra(EXTRA_EXTRA, originalDownloadInfo.getExtras());
+        }
+        context.sendBroadcast(intent);
+    }
+
+    private String getPackageName() {
+        return context.getApplicationContext().getPackageName();
+    }
+
     private void notifyThroughDatabase(State state, int finalStatus, String errorMsg, int numFailed) {
-        originalDownloadInfo.setStatus(finalStatus);
```

Comment:
it holds it, but it only gets updated via the database, the download thread also never requeried the status. 

Now we only update the database entry for that download